File: days/day1.js
const fs = require("fs")

const data = fs.readFileSync("../input/input1.txt", "utf-8").split('\n').map(elem => parseInt(elem))

//--------------------------------------------------------------------------------------\\

let last = 0;
let increases = 0;
for (let i of data) {
    if (last !== 0 && i > last) {
        increases++;
    }
    last = i;
}
console.log(increases);

//--------------------------------------------------------------------------------------\\

let sum = data[0] + data[1] + data[2]
let increases2 = 0;
for (let i = 1; i < data.length - 2; i++) {
    let newSum = data[i] + data[i + 1] + data[i + 2]
    if (newSum > sum) {
        increases2++
    }
    sum = newSum
}
console.log(increases2)

File: days/day1.go
package main

import (
    "bufio"
    "fmt"
    "log"
    "os"
    "strconv"
)

func main() {
    lines := readFile()
    dataSet := resultToInt(lines)
    last := 0
    increseas := 0
    for _, data := range dataSet {
        if last != 0 {
            if data > last {
                increseas++
            }
        }
        last = data
    }
    fmt.Println(increseas)
}

func resultToInt(input []string) (result []int) {
    for _, i := range input {
        integer, _:= strconv.Atoi(i)
        result = append(result, integer)
    }
    return
}

func readFile() (input []string ){
    file, err := os.Open("../input/input1.txt")
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()

    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        input = append(input, scanner.Text())
    }

    if err := scanner.Err(); err != nil {
        log.Fatal(err)
    }
    return
}